x/ecocredit/server/marketplace/msg_remove_allowed_denom.go

```diff
@@ -0,0 +1,33 @@
+package marketplace
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/orm/types/ormerrors"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+// RemoveAllowedDenom removes denom from the allowed denoms.
+func (k Keeper) RemoveAllowedDenom(ctx context.Context, req *marketplace.MsgRemoveAllowedDenom) (*marketplace.MsgRemoveAllowedDenomResponse, error) {
+	if k.authority.String() != req.Authority {
+		return nil, govtypes.ErrInvalidSigner.Wrapf("invalid authority: expected %s, got %s", k.authority, req.Authority)
+	}
+
+	allowedDenom, err := k.stateStore.AllowedDenomTable().Get(ctx, req.Denom)
+	if err != nil {
+		if ormerrors.NotFound.Is(err) {
+			return nil, sdkerrors.ErrNotFound.Wrapf("allowed denom %s not found", req.Denom)
```

Comment:
```suggestion
			return nil, sdkerrors.ErrNotFound.Wrapf("allowed denom %s", req.Denom)
```

can prob remove the "not found" string, since it will already have it in the parent error